components/renku_data_services/session/shipwright_core.py

```diff
@@ -0,0 +1,111 @@
+"""Business logic for ShipWright resources."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from renku_data_services.session import constants, crs, models
+from renku_data_services.session.k8s_client import ShipwrightClient
+
+if TYPE_CHECKING:
+    from renku_data_services.app_config.config import BuildsConfig
+
+
+async def update_build_status(
+    build: models.Build, shipwright_client: ShipwrightClient | None
+) -> models.ShipWrightBuildStatusUpdate | None:
+    """Update the status of a build by pulling the corresponding BuildRun from ShipWright.
+
+    Note: this method will update `build` in place.
+    """
+    if shipwright_client is None:
+        logging.warning("ShipWright client not defined, BuildRun refresh skipped.")
+        return None
+
+    k8s_name = build.get_k8s_name()
+    k8s_build = await shipwright_client.get_build_run(name=k8s_name)
+
+    if k8s_build is None:
+        return models.ShipWrightBuildStatusUpdate(status=models.BuildStatus.failed)
+
+    k8s_build_status = k8s_build.status
+    completion_time = k8s_build_status.completionTime if k8s_build_status else None
+
+    if k8s_build_status is None or completion_time is None:
+        return None
+
+    conditions = k8s_build_status.conditions
+    condition = next(filter(lambda c: c.type == "Succeeded", conditions or []), None)
+
+    buildSpec = k8s_build_status.buildSpec
+    output = buildSpec.output if buildSpec else None
+    result_image = output.image if output else "unknown"
+
+    source = buildSpec.source if buildSpec else None
+    git_obj = source.git if source else None
+    result_repository_url = git_obj.url if git_obj else "unknown"
+
+    source_2 = k8s_build_status.source
+    git_obj_2 = source_2.git if source_2 else None
+    result_repository_git_commit_sha = git_obj_2.commitSha if git_obj_2 else None
+    result_repository_git_commit_sha = result_repository_git_commit_sha or "unknown"
+
+    if condition is not None and condition.status == "True":
+        return models.ShipWrightBuildStatusUpdate(
+            status=models.BuildStatus.succeeded,
+            completed_at=completion_time,
+            result=models.BuildResult(
+                completed_at=completion_time,
+                image=result_image,
+                repository_url=result_repository_url,
+                repository_git_commit_sha=result_repository_git_commit_sha,
+            ),
+        )
+    else:
+        return models.ShipWrightBuildStatusUpdate(
+            status=models.BuildStatus.failed,
+            completed_at=completion_time,
+        )
+
+
+async def create_build(
+    build: models.Build,
```

Comment:
suggestion: The only thing you need from the build is the name. So only take that. Rename this to `name: str`. The same applies to all the other functions in this file.

Reply:
Done in [705fef2](https://github.com/SwissDataScienceCenter/renku-data-services/pull/625/commits/705fef2a1e8db539f9678d57f309ceb8ecc9eff9).